src/layouts/BasicLayout.js

```diff
@@ -98,6 +98,16 @@ class BasicLayout extends React.PureComponent {
     }
     return title;
   }
+  getBashRedirect = () => {
+    // According to the url parameter to redirect
+    // 这里是重定向的,重定向到 url 的 redirect 参数所示地址
+    const urlParams = new URL(window.location.href);
+    const redirect = urlParams.searchParams.get('redirect') || '/dashboard/analysis';
+    // Remove the parameters in the url
+    urlParams.searchParams.delete('redirect');
+    window.history.pushState(null, 'redirect', urlParams.href);
```

Comment:
avoid using window.  check this.prop.history.push   - doesnt react router Redirect handle the push?